gcloud-java-compute/src/main/java/com/google/gcloud/compute/NetworkInterface.java

```diff
@@ -0,0 +1,505 @@
+ */
+public class NetworkInterface implements Serializable {
+
+  static final Function<com.google.api.services.compute.model.NetworkInterface, NetworkInterface>
+      FROM_PB_FUNCTION =
+      new Function<com.google.api.services.compute.model.NetworkInterface,
+          NetworkInterface>() {
+        @Override
+        public NetworkInterface apply(
+            com.google.api.services.compute.model.NetworkInterface pb) {
+          return NetworkInterface.fromPb(pb);
+        }
+      };
+  static final Function<NetworkInterface, com.google.api.services.compute.model.NetworkInterface>
+      TO_PB_FUNCTION =
+      new Function<NetworkInterface,
+          com.google.api.services.compute.model.NetworkInterface>() {
+        @Override
+        public com.google.api.services.compute.model.NetworkInterface apply(
+            NetworkInterface networkInterface) {
+          return networkInterface.toPb();
+        }
+      };
+
+  private static final long serialVersionUID = 936741262053605581L;
+
+  private final String name;
+  private final NetworkId network;
+  private final String networkIp;
+  private final SubnetworkId subnetwork;
+  private final List<AccessConfig> accessConfigurations;
+
+  /**
+   * Access configuration for a Google Compute Engine instance's network interface. Objects of this
+   * class can be used to assign either a static or an ephemeral external IP address to Google
+   * Compute Engine instances.
+   *
+   * @see <a href="https://cloud.google.com/compute/docs/instances-and-network#reservedaddress">
+   *     Static external IP addresses</a>
+   * @see <a href="https://cloud.google.com/compute/docs/instances-and-network#ephemeraladdress">
+   *     Ephemeral external IP addresses</a>
+   */
+  public static final class AccessConfig implements Serializable {
+
+    static final Function<com.google.api.services.compute.model.AccessConfig, AccessConfig>
+        FROM_PB_FUNCTION =
+        new Function<com.google.api.services.compute.model.AccessConfig, AccessConfig>() {
+          @Override
+          public AccessConfig apply(com.google.api.services.compute.model.AccessConfig pb) {
+            return AccessConfig.fromPb(pb);
+          }
+        };
+    static final Function<AccessConfig, com.google.api.services.compute.model.AccessConfig>
+        TO_PB_FUNCTION =
+        new Function<AccessConfig, com.google.api.services.compute.model.AccessConfig>() {
+          @Override
+          public com.google.api.services.compute.model.AccessConfig apply(
+              AccessConfig instance) {
+            return instance.toPb();
+          }
+        };
+
+    private static final long serialVersionUID = -5438060668934041567L;
+
+    private final String name;
+    private final String natIp;
+    private final Type type;
+
+    /**
+     * The type of network access configuration. The only supported value is {@code ONE_TO_ONE_NAT}.
+     */
+    public enum Type {
+      ONE_TO_ONE_NAT
```

Comment:
I could find any in the service docs but i think it refers to the networking concept of 1:1 NAT. Which (very roughly speaking) is assigning a public address to a machine so that it can be reached from the outside.